packages/components/charts/src/chart-tooltip.tsx

```diff
@@ -14,7 +14,8 @@ export const ChartTooltip = forwardRef<ChartTooltipProps, "div">(
     const { styles } = useTooltip()
 
     const items = payload.map(
-      ({ color, name, value: valueProp } = {}, index) => {
+      ({ color: colorProp, name, value: valueProp, ...rest } = {}, index) => {
+        const color = colorProp ?? rest.payload.color
```

Comment:
When does `rest.payload.color` exist?

If it exists, `{...rest}` will insert it into the tag as an unnecessary attribute.

Reply:
If `tooltipDataSource` is `segment`, the color is passed to `payload.color` instead of `color`.
The default value of `tooltipDataSource` is `all`.
```tsx
<DonutChart tooltipDataSource="segment" />
```